### DISCUSSION TITLE
How many max total schemas?

### DISCUSSION THREAD

**Discussion by woile** [Apicurio/apicurio-registry#4398] [Q&A] [answered]
I'm doing an investigation on schema registries. I'm trying to find out what's the hard limit on apicurio. 
According to the [settings](https://www.apicur.io/registry/docs/apicurio-registry/2.3.x/getting-started/assembly-configuring-the-registry.html) there's a `Max total schemas` which by default is `-1`, which would lead to believe that you can have as many as u want.
Looking at the confluent schema registry, someone did a [benchmark](https://github.com/thilo-behnke/schema-registry-benchmark) because it holds all schemas in memory. So the number of schemas depends on the memory you have. 
Anyone knows how apicurio compares to the confluent schema-registry? 

Thanks!

**Comment by EricWittmann** [comment, marked as answer]
Hi @woile.  The answer to this question depends (to some degree) on the storage mechanism you choose.  We support using either a conventional DB or Kafka as the persistent store.  If the chosen storage is a DB (e.g. postgresql), then the max number of schemas is essentially unbounded (practically bounded only by the normal limits of an indexed Postgresql table).

If you choose Kafka as the storage, then you are bounded by memory, much like the Confluent solution.  We keep all of the schemas and metadata in memory, and so you might choose to set whatever max you find sensible (this will depend on how big your schemas are, on average).

We assume the number of schemas for most use-cases will tend to be small (in the 1000s at most?) so either solution is typically fine in practice.
